### PR TITLE
Gasbot caching

### PR DESCRIPTION
**Pull request by nicholaspai**
Add `update()` wrapper to `gasEstimator` similarly to `ExpiringMultiPartyClient`. 
Follow up to #1179

### CODE REVIEW

File: disputer/disputer.js [hunk 1]
```diff
@@ -18,19 +18,21 @@ class Disputer {
...
       message: "Checking for any disputable liquidations"
     });
 
-    // Update the client to get the latest liquidation information.
-    await this.empClient._update();
-
-    // Update the gasEstimator to get the latest gas price data.
-    // If the client has a data point in the last 60 seconds returns immediately.
-    await this.gasEstimator._update();
+    await this.update();
```
Comment (mrice32) [inline disputer/disputer.js:35]: Looks like this PR has repeated changes from past PRs? Maybe rebase your most recent commit onto master?

Comment (nicholaspai) [reply disputer/disputer.js]: @mrice32 i forgot to merge master, check now!

File: liquidator/test/Liquidator.js [hunk 2]
```diff
@@ -296,8 +296,10 @@ contract("Liquidator.js", function(accounts) {
 
     // liquidatorBot creates a position to have synthetic tokens to pay off debt upon liquidation.
     await emp.create({ rawValue: toWei("1000") }, { rawValue: toWei("500") }, { from: liquidatorBot });
+    // No need to force update the `empClient` here since we are not interested in detecting the `liquidatorBot`'s new position,
+    // but now when we try to liquidate the position the liquidation will go through because the bot will have the requisite balance.
 
-    // No transaction should be sent, so this should not throw.
+    // Can now liquidate the position.
```
Comment (mrice32) [inline liquidator/test/Liquidator.js:302]: Is this change supposed to be here?

Comment (nicholaspai) [reply liquidator/test/Liquidator.js:302]: yeah, shouldn't the `liquidator` now be able to liquidate the position?  The liquidator creates tokens in order to liquidate and we check on line `306` that the liquidation exists

Comment (mrice32) [reply liquidator/test/Liquidator.js:302]: SGTM, just wanted to make sure it was meant for this PR vs being an artifact from the previous one.

**Pull request merged**